pom.xml

```diff
@@ -183,7 +184,7 @@
         <dep.accumulo-hadoop.version>2.7.7-1</dep.accumulo-hadoop.version>
         <dep.accumulo.version>3.0.0</dep.accumulo.version>
         <dep.airlift.version>248</dep.airlift.version>
-        <dep.alluxio.version>312</dep.alluxio.version>
+        <dep.alluxio.version>2.9.4</dep.alluxio.version>
```

Comment:
I'm confused by the Alluxio versioning scheme... [Maven central says `2.9.4` was released after `312`](https://mvnrepository.com/artifact/org.alluxio/alluxio-core-common), but `312` was released after `2.9.3`.

Are the `x.x.x` and `xxx` versioning schemes completely separate? Should we use one or the other in particular?

(Trying to make sure this isn't a downgrade that could affect the Alluxio cache fs)

Reply:
@mosabua who can answer this?

Reply:
they switched from semver to full numbers so we need to stick with 3xx .. 2.9.4 is the old incompatible stuff and I guess they continue to release backports of some stuff.

Reply:
So for the filesystem caching we already have implemented it needs to stay with 312 or 3xx or newer from all I know. 

For the new filesystem module I think it would be best to also use that newer version .. but it might not be possible (yet) .. if that is the case we need to probably use two different properties

Reply:
2.9.4 is a more stable version so we would keep backporting changes from 3xx to 2.9.x line

Reply:
We can also pull in Alluxio experts if needed.

Reply:
Pretty sure we needed to use 3xx for some fixes that went into it .. how do we know everything was backported that is needed? Can we maybe separate the concerns for this new file system support vs the existing file system caching ?

Reply:
see https://github.com/trinodb/trino/pull/21038

Reply:
If the CVE fix is not backported .. we should NOT use 2.9.4

Reply:
https://github.com/Alluxio/alluxio/commit/ac4b12c0978985b7b099a6cf036cd6924b0b9c1b it's in 2.9.4
For any change Trino side want to fix in Alluxio we are more than happy to fix it and guarantee to keep it in sync with 2.9.x line since 2.9.x line is still the stable version.

Reply:
Okay ... this just means that we are doing a downgrade in terms of dependency then .. and that we need to somehow make sure the file system caching is not affected. 

fyi @electrum @raunaqmorarka @Pluies

Reply:
https://github.com/Alluxio/alluxio/commit/3e520f1fef3b79b488088c339bddc7bd321d333d we have a sync to keep Trino using the same filesystem cache code between 3xx and 2.9.4. We also go through some internal testing to make sure the functionality is working as expected. Is there any more concerns from Trino side? We want to hear from you guys and solve it together.

Reply:
The only concern I have is that in the original intro of Alluxio for fs caching we somehow decided we needed to move to 3xx. From what I recall on info from Alluxio. If we now go back .. thats fine as long as we dont break any existing functionality. But thats just my 2c. If 2.x is stable and 3xx is newer/unstable .. it probably makes sense to go back to 2.x

Reply:
cc @amoghmargoor

Reply:
If a downgrade in version is really desirable, we should just do it as a separate PR first.

Reply:
Good idea.. could really be in parallel in terms of timing and still be used in this PR as well. Then we could verify separately and also just merge the new PR first, rebase this, and ship this.

Reply:
SGTM as well, I would submit a separate PR to downgrade the version first

Reply:
https://github.com/trinodb/trino/pull/22350 Hi @mosabua , I submit a PR to downgrade alluxio version as we discussed. In the meantime, we still have issue with the docker test. Do you think we can have someone help us on this or just give us some hint on what might go wrong? https://github.com/trinodb/trino/pull/21603#issuecomment-2123536879